### PR TITLE
feat(coderd/database): use `template_usage_stats` in `GetTemplateAppInsights` query

### PR DESCRIPTION
**Pull request by mafredri**
This PR updates the `GetTemplateAppInsights` query to use rolled up `template_usage_stats` instead of raw agent and app stats.

### CODE REVIEW

**Comment by mafredri** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/coder/coder/12669?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#12670** <a href="https://app.graphite.dev/github/pr/coder/coder/12670?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12669** <a href="https://app.graphite.dev/github/pr/coder/coder/12669?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#12668** <a href="https://app.graphite.dev/github/pr/coder/coder/12668?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12667** <a href="https://app.graphite.dev/github/pr/coder/coder/12667?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12666** <a href="https://app.graphite.dev/github/pr/coder/coder/12666?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12665** <a href="https://app.graphite.dev/github/pr/coder/coder/12665?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12664** <a href="https://app.graphite.dev/github/pr/coder/coder/12664?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mafredri and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by johnstcn — APPROVED** [review]

File: coderd/database/queries/insights.sql [hunk 1]
```diff
@@ -182,57 +182,93 @@ GROUP BY template_id;
...
-		w.id = was.workspace_id
-		AND CASE WHEN COALESCE(array_length(@template_ids::uuid[], 1), 0) > 0 THEN w.template_id = ANY(@template_ids::uuid[]) ELSE TRUE END
-	)
-	-- We do a left join here because we want to include user IDs that have used
-	-- e.g. ports when counting active users.
-	LEFT JOIN workspace_apps wa ON (
-		wa.agent_id = was.agent_id
-		AND wa.slug = was.slug_or_port
+WITH
+	app_insights AS (
```
Comment (mtojek) [inline coderd/database/queries/insights.sql:186]: nit: you may want to drop a comment here, many joins around

**Review by mtojek — APPROVED** [review]

File: coderd/database/queries/insights.sql (hunk 1 above)
Comment (mafredri) [reply coderd/database/queries/insights.sql:186]: Hmm, in my opinion it's a self-contained sequence of joins and there is a comment inline, is there something more you would like explained in the comment? This is a common pattern when we need to lookup agent (or in our case app) -> template association.

Comment (mtojek) [reply coderd/database/queries/insights.sql:186]: Yeah, exactly this :)

prevent going through joins and "fast-forward" reading query code by human

**Comment by mafredri** [comment]
### Merge activity

* **Mar 25, 9:33 AM EDT**: @mafredri started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/coder/coder/12669).

**Pull request merged**